### PR TITLE
Add geoip data

### PR DESCRIPTION
**Pull request by tspenov**
## Changes

Save geoip data for PPP implementation

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/geoip/data.ex [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+            changeset =
+              changeset(%__MODULE__{}, %{
+                ip_address: remote_ip,
+                is_vpn: is_vpn,
+                country_name: data["country_name"],
+                country_code: data["country_code2"]
+              })
+
+            Repo.insert(changeset)
```
Comment (IvanIvanoff) [inline lib/sanbase/geoip/data.ex:40]: nitpick: extract into a function

File: lib/sanbase/geoip/geoip.ex [hunk 2]
```diff
@@ -0,0 +1,44 @@
+defmodule Sanbase.Geoip do
+  require Logger
+
+  def fetch_geo_data(ip) do
+    tag = "Geoip.fetch_geo_data"
+    url = "https://api.ipgeolocation.io/ipgeo?ip=#{ip}&include=security"
+    headers = [{"Origin", "https://ipgeolocation.io"}]
+
+    case HTTPoison.get(url, headers, timeout: 1000) do
```
Comment (IvanIvanoff) [inline lib/sanbase/geoip/geoip.ex:9]: Isn't 1000ms too low? What will be the behavior of the FE if this call fails?

File: lib/sanbase_web/graphql/resolvers/billing_resolver.ex [hunk 3]
```diff
@@ -10,7 +10,20 @@ defmodule SanbaseWeb.Graphql.Resolvers.BillingResolver do
...
+          origin_host_parts: origin_host_parts,
+          remote_ip: remote_ip
+        }
+      }) do
+    Logger.info(
+      "products_with_plans: #{inspect(origin_url)}, #{inspect(origin_host_parts)}, #{inspect(remote_ip)}"
+    )
+
+    remote_ip = Sanbase.Utils.IP.ip_tuple_to_string(remote_ip)
+    Sanbase.Geoip.Data.find_or_insert(remote_ip)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/billing_resolver.ex:25]: The return from here is not used, is this expected?

File: priv/repo/migrations/20231206123012_create_geoip_data.exs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+defmodule Sanbase.Repo.Migrations.CreateGeoipData do
+  use Ecto.Migration
+
+  def change do
+    create table(:geoip_data) do
+      add(:ip_address, :string)
+      add(:is_vpn, :boolean)
+      add(:country_name, :string)
+      add(:country_code, :string)
```
Comment (IvanIvanoff) [inline priv/repo/migrations/20231206123012_create_geoip_data.exs:9]: The changeset defines all fields as required. You can put `null: false` here.

File: lib/sanbase/geoip/data.ex [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    |> validate_required([:ip_address, :is_vpn, :country_name, :country_code])
+  end
+
+  def find_or_insert(remote_ip) do
+    case Repo.get_by(__MODULE__, ip_address: remote_ip) do
+      nil ->
+        case Geoip.fetch_geo_data(remote_ip) do
+          {:ok, data} ->
+            is_vpn =
+              data["security"]["is_proxy"] == true and data["security"]["proxy_type"] == "VPN"
```
Comment (IvanIvanoff) [inline lib/sanbase/geoip/data.ex:30]: Can it be `is_proxy == false` but a VPN? Is the first check really needed?

File: lib/sanbase_web/graphql/resolvers/billing_resolver.ex (hunk 3 above)
Comment (tspenov) [reply lib/sanbase_web/graphql/resolvers/billing_resolver.ex:25]: Yes, for beginning just wanted to save and see some stats

File: lib/sanbase/geoip/data.ex (hunk 5 above)
Comment (tspenov) [reply lib/sanbase/geoip/data.ex:30]: No,  when`is_proxy == false` , proxy_type couldn't be VPN.

File: lib/sanbase/geoip/geoip.ex (hunk 2 above)
Comment (tspenov) [reply lib/sanbase/geoip/geoip.ex]: Added 3s timeout

**Pull request merged**